### DISCUSSION TITLE
リリースまでにすること

### DISCUSSION THREAD

**Discussion by Uni-21** [nam-club/NumBuyer_front#110] [General]
リリースまでにすることを書いてく
スレッド毎1テーマで

**Comment by aoking1213** [comment]
@UniTofu
update_stateにターン情報追加よな

**Reply by Uni-21** [reply]
> ターンの表記はあったほうがよいかも
ユーザとしては今何ターンかわからないため

next_turnでも良いかも

**Reply by aoking1213** [reply]
@UniTofu 
あ、そうだった
next_turnあったな

**Reply by Uni-21** [reply]
next_turnのレスポンスにcurrentTurnを追加
```
playerId:(int)
cards[ ]:(String) 
coin:(int)
targetCard:(String)　←正解者が出た時に更新
auctionCards[ ]:(String)
currentTurn:(int)
```

**Reply by Uni-21** [reply]
push済み
https://github.com/nam-club/NumBuyer_back/commit/5645b51ed785bfd283e53d6d8748defb9d9e5c09

**Comment by Uni-21** [comment]
@aoking1213 
ドメイン、`nam-club.com` で良い？
numbuyerは `numbuyer.nam-club.com` になる。
今後割合のアプリがリリースされるなら、例えば↓のような感じになる。
`wariai.nam-club.com`

**Reply by Uni-21** [reply]
`nam-club.com/numbuyer/`とする方法についてメモ

↓ の方法はなかなか大変そう。
https://qiita.com/ShortArrow/items/3b8960d51b50fd198524

↓ [React Rotue](https://reactrouter.com/)を使うことでトップの相対パスはできたが、Lobbyができなかった。
https://qiita.com/backpaper0@github/items/113f8a3304ffd4d4d30c

**Reply by Uni-21** [reply]
今更ながら気づいたが、パスによるルーティングはできなそう。

例えば以下の
`www.nam-club.com/numbuyer/`
`www.nam-club.com/wariai/`
があるとしても、お名前.comで設定できるのは`www.nam-club.com`に対するルーティングのみ。リクエストに対しどちらかのパスに転送する、といったことはできない。

なのでパスではなくサブドメインによって分ける方法で対応。
https://numbuyer.nam-club.com/

ちなみにサブドメインによって分ける方法は一般的にある。
```
・Yahoo!JAPAN ： https://www.yahoo.co.jp/
・Yahoo!ショッピング ： https://shopping.yahoo.co.jp/
・Yahoo!プレミアム ： https://premium.yahoo.co.jp/
・ヤフオク ： https://auctions.yahoo.co.jp/
・Yahoo!ニュース ： https://news.yahoo.co.jp/
```

**Comment by Uni-21** [comment]
仕様こうしたほうがいいかも、と思ったことについて書いていくスレッド
あくまでディスカッションなので、この中から必要と思ったとこだけissueに積んで対応する

**Reply by Uni-21** [reply]
・COM対戦機能
adsenseの審査2度やったが通らないしやはり最低限の機能を持ったCOM対戦機能は作ったほうが良いのかも・・
トップページにCOM Matchのようなボタンを追加するイメージ。
細かい機能はつけずまずは最小限で。
4人対戦、COMが自動で3人作られる,COMの強さは固定

=> 必要

**Reply by Uni-21** [reply]
（超優先度は低いし、なんならなくても良いけど）
効果音あっても良さそう

**Reply by Uni-21** [reply]
トップページに追加したほうが良いかもなやつ
- 共有ボタン
- タイトルの下に超簡単で小さめの説明（パッと開いた時になんのサイトかわからないため)

**Reply by Uni-21** [reply]
・404ページ
不正なurlの時netlifyのリンクが表示されてるのでよくなさそう。
トップに戻すか、404を表示するのが望ましい
<img width="835" alt="Screen Shot 2022-10-05 at 2 34 07" src="https://user-images.githubusercontent.com/51052544/193887580-25b25fbc-0183-44b7-86bf-ce2caa296d72.png">

**Reply by Uni-21** [reply]
google検索に表示されるようになったがnetlifyデフォルトの説明 & 日本語でない
<img width="888" alt="Screen Shot 2022-10-07 at 0 23 31" src="https://user-images.githubusercontent.com/51052544/194353531-e00a1946-7906-4528-a920-73cfc51dddff.png">
